src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile_strategy.py

```diff
@@ -233,3 +236,19 @@ def extra_compile_classpath_iter():
           yield (conf, jar)
 
     return list(extra_compile_classpath_iter())
+
+  # Work in a tmpdir so we don't stomp the main analysis files on error.
```

Comment:
This comment should go in a pydoc  ( use `""" description """` on the line below the method definition)